### PR TITLE
feat(processors.round): Add plugin

### LINKED ISSUE #17077: Add denoise plugin
### Use Case

For values that are prone to fluctuation, sometimes it is not useful to record a datapoint for very minor changes in the value. This plugin would allow for rounding these values to the provided number of significant figures. Combined with the `dedup` processor it would then be possible to prevent values which are very similar from being sent to an output.

### Expected behavior

A plugin is available that can be configured to perform this de-noising.

### Actual behavior

This plugin is not currently available.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by alexgokhale**
## Summary

This PR adds a `denoise` plugin, which allows numeric field values to be rounded to the specified number of significant figures. This is particularly useful for metrics that fluctuate often, as after they have been rounded they can be deduplicated effectively.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #17077

### CODE REVIEW

File: plugins/processors/denoise/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Denoise Processor Plugin
+
+The _Denoise_ processor is used to denoise to numerical field values. For each
+field, the value will be truncated to the specified number of significant
+figures. This can be particularly useful in combination with the `dedup`
+processor to filter out values which are constantly fluctuating.
```
Comment (srebhan) [inline plugins/processors/denoise/README.md]: We do have a kind of fixed pattern for the intro and we also need metadata for generating the docs so how about

```suggestion
This plugin allows to round numerical field values to the configured precision. This is
particularly useful in combination with the [dedup processor][dedup] to reduce the number of
metrics sent to the output if only a lower precision is required for the values.

⭐ Telegraf v1.36.0
🏷️ math
💻 all

[dedup]: /plugins/processors/dedup/README.md
```

File: plugins/processors/denoise/sample.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Remove noise from numerical fields
+[[processors.denoise]]
```
Comment (srebhan) [inline plugins/processors/denoise/sample.conf]: Can we please name the processor `round` as this is what it actually does...

File: plugins/processors/denoise/sample.conf [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  ## Significant figures to trim to
+  # sf = 3
```
Comment (srebhan) [inline plugins/processors/denoise/sample.conf]: How about naming it `precision` with a positive number meaning rounding to the left of the decimal separator and negative means rounding to the fraction i.e. right of the decimal separator?

File: plugins/processors/denoise/sample.conf [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+  ## Denoise only numeric fields matching the filter criteria below.
+  ## Excludes takes precedence over includes.
+  # include_fields = []
+  # exclude_fields = []
```
Comment (srebhan) [inline plugins/processors/denoise/sample.conf]: As a convention we do have
```suggestion
  # fields_include = []
  # fields_exclude = []
```

File: plugins/processors/denoise/denoise.go [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+const (
+	defaultSignificantFigures = 3
+)
```
Comment (srebhan) [inline plugins/processors/denoise/denoise.go]: Please directly use this value in `init` without defining a constant.

File: plugins/processors/denoise/denoise.go [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+type Denoise struct {
+	SignificantFigures int             `toml:"sf"`
+	IncludeFields      []string        `toml:"include_fields"`
+	ExcludeFields      []string        `toml:"exclude_fields"`
+	Log                telegraf.Logger `toml:"-"`
+	fieldFilter        filter.Filter
```
Comment (srebhan) [inline plugins/processors/denoise/denoise.go]:
```suggestion
	Log                telegraf.Logger `toml:"-"`
	
	fields filter.Filter
```

File: plugins/processors/denoise/denoise.go [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		if v == 0 {
+			return float64(0)
+		}
+
+		return roundToSignificantFigures(v, p.SignificantFigures)
+	default:
+		p.Log.Debugf("Value (%v) type invalid: [%v] is not an int, uint or float", v, reflect.TypeOf(value))
+	}
+	return value
+}
```
Comment (srebhan) [inline plugins/processors/denoise/denoise.go]: Those functions should be moved to the end of the file before `init`...

Furthermore, I think you should keep the type (i.e. int, uint, float) of the field... How about

```suggestion
func (p *Denoise) Apply(metrics ...telegraf.Metric) []telegraf.Metric {
	// We might want to move this into Init()
	factor := math.Pow10(p.Precision)

	for _, metric := range metrics {
		for _, field := range metric.FieldList() {
			if !p.fieldFilter.Match(field.Key) {
				continue
			}

			switch v := field.Value.(type) {
			case int:
				field.Value = roundInt(v, int64(factor))
			case int8:
				field.Value = roundInt(v, int64(factor))
			case int16:
				field.Value = roundInt(v, int64(factor))
			case int32:
				field.Value = roundInt(v, int64(factor))
			case int64:
				field.Value = roundInt(v, int64(factor))
			case uint:
				field.Value = roundInt(v, int64(factor))
			case uint8:
				field.Value = roundInt(v, int64(factor))
			case uint16:
				field.Value = roundInt(v, int64(factor))
			case uint32:
				field.Value = roundInt(v, int64(factor))
			case uint64:
				field.Value = roundInt(v, int64(factor))
			case float32:
				field.Value = roundFloat(v, factor)
			case float64:
				field.Value = roundFloat(v, factor)
		}
	}
	return metrics
}

func roundInt[V constraints.Integer](value V, factor int64) interface{} {
	f := V(factor)

	if factor < 10 {
		return value
	}

	v := value / f

	remainder := (value / (f / 10)) % 10
	if remainder < 5 {
		return v * f
	}

	return (v + 1) * f
}

func roundFloat[V constraints.Float](value V, factor float64) V {
	return V(math.Round(float64(value)/factor) * factor)
}
```

File: plugins/processors/denoise/denoise_test.go [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+const (
+	epsilon = float64(0.00000001)
+)
```
Comment (srebhan) [inline plugins/processors/denoise/denoise_test.go]:
```suggestion
const epsilon = float64(0.00000001)
```
or directly use the value in the test instead of defining a constant.

File: plugins/processors/denoise/denoise_test.go [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+func TestDenoise(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    []telegraf.Metric
+		expected []telegraf.Metric
+	}{
+		{
+			name: "int64",
+			input: []telegraf.Metric{
+				testutil.MustMetric("cpu",
```
Comment (srebhan) [inline plugins/processors/denoise/denoise_test.go]: Please use `metric.New` instead. We want to get rid of `testutil.MustMetric` at some point as it basically duplicates `metric.New`...

Same for all other instances...

**Comment by alexgokhale** [comment]
Thanks for the review! I was away last week but will actions your comments as soon as possible :)

File: plugins/processors/denoise/denoise_test.go (hunk 9 above)
Comment (alexgokhale) [reply plugins/processors/denoise/denoise_test.go]: Actioned in 76725ac

File: plugins/processors/denoise/denoise_test.go (hunk 8 above)
Comment (alexgokhale) [reply plugins/processors/denoise/denoise_test.go]: Actioned in 76725ac

File: plugins/processors/denoise/denoise.go (hunk 6 above)
Comment (alexgokhale) [reply plugins/processors/denoise/denoise.go]: Actioned in 76725ac

File: plugins/processors/denoise/denoise.go (hunk 5 above)
Comment (alexgokhale) [reply plugins/processors/denoise/denoise.go]: Actioned in 76725ac

File: plugins/processors/denoise/sample.conf (hunk 4 above)
Comment (alexgokhale) [reply plugins/processors/denoise/sample.conf]: Actioned in 76725ac

File: plugins/processors/denoise/sample.conf (hunk 2 above)
Comment (alexgokhale) [reply plugins/processors/denoise/sample.conf]: Actioned in 76725ac

File: plugins/processors/denoise/README.md (hunk 1 above)
Comment (alexgokhale) [reply plugins/processors/denoise/README.md]: Actioned in 76725ac

**Comment by srebhan** [comment]
@alexgokhale please let me know once you are ready for another review round...

**Comment by alexgokhale** [comment]
@srebhan Hey -- just wanted to check in before I make the last changes (to switch to Precision and then re-organise the file):

The use case that prompted this feature request from myself was for different voltage metrics being captured -- one is a 24V line and another is a 5V line. In both cases we want 3 s.f., but this would be a different number of decimal places for each (e.g. 23.8V for the 24V rail, 5.21V for the 5V rail).

I appreciate that when creating a processor for Telegraf the desire is to build for what will be most useful across all cases, so I think it would be reasonable to switch to Precision instead -- would the recommended flow for a use case like my own be to have two different instances of the processor (one with precision `-1`, the other with `-2`, each targeting different metrics/fields?

**Comment by alexgokhale** [comment]
Also one clarification on the meaning of positive/negative numbers on precision &ndash; might it make more sense to have a positive number meaning to the right of the decimal separator and negative meaning to the left?

I think this would probably better reflect the common use cases for a plugin like this, which are (probably) more likely to be rounding to a number of decimal places than to be rounding large numbers to a certain number of significant figures?

One potential risk of the precision concept as a whole is that it could be very difficult to define for larger numbers &ndash; for example if you're dealing with values in the millions/billions range, it could be more intuitive to define the rounding by the number of significant figures, rather than having to provide a precision of 7 for example.

E.g. for a metric with value 12345678, providing 2 and getting an output of 12000000 might be clearer than needing to provide 7, especially if the numbers are going to be on the border of rolling over/under to a new digit.

Hopefully this makes sense, happy to clarify anything but thought it could be useful to lay out how I've been thinking about usage so far! 😅

**Comment by srebhan** [comment]
> [...] would the recommended flow for a use case like my own be to have two different instances of the processor (one with precision -1, the other with -2, each targeting different metrics/fields?

Yes. Alternatively you could change the processor such that you can specify multiple "subsections" with a precision and a field filter to stay with one instance. However, this would probably make things more complicated for other people as the would need another subsection.

One additional question regarding your use-case: Why do you round in Telegraf and not on the query/visualization side?

**Comment by srebhan** [comment]
> Also one clarification on the meaning of positive/negative numbers on precision – might it make more sense to have a positive number meaning to the right of the decimal separator and negative meaning to the left?

Yeah you are right. I was thinking of it as `10^precision` that's why I used this nomenclature but your mileage may vary... :-) I'm fine either way as long as the setting description reflects what it does.

**Comment by srebhan** [comment]
> One potential risk of the precision concept as a whole is that it could be very difficult to define for larger numbers – for example if you're dealing with values in the millions/billions range, it could be more intuitive to define the rounding by the number of significant figures, rather than having to provide a precision of 7 for example.

I don't agree as usually you do have use-case specific requirements e.g. I need everything in a precision of 1/10th Volt or 1kV for my application. It seems like your use-case is more "visualization" driven and I would really suggest to do that type of manipulation on the query/visualization side...

**Comment by alexgokhale** [comment]
> Alternatively you could change the processor such that you can specify multiple "subsections" with a precision and a field filter to stay with one instance. However, this would probably make things more complicated for other people as the would need another subsection.

I agree &ndash; will leave as is, I think it's reasonable to just use more than one processor.

> Why do you round in Telegraf and not on the query/visualization side?

We collect these values from a number of devices in a fleet, and naturally they can be very noisy (due to the precision we capture them at). We use the `dedup` processor to prevent us collecting values that haven't changed for other metrics, which significantly reduces the amount of data we need to store, but this currently isn't possible for our voltage/other noisy values because they fluctuate so much.

Because we only care about understanding these values to a certain precision (i.e. it's not useful to have the voltage to 10 d.p., nor is it probably that accurate), it would be ideal to avoid having to ingest this data in the first place.

> I need everything in a precision of 1/10th Volt or 1kV for my application.

Good point, I think we're aligned on implementation 👍

File: plugins/processors/round/README.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
+# Round Processor Plugin
+
+This plugin allows to round numerical field values to the configured precision.
+This is particularly useful in combination with the [dedup processor][dedup]
+to reduce the number of metrics sent to the output if only a lower precision
+is required for the values.
+
+⭐ Telegraf v1.36.0
+🏷️ math
```
Comment (alexgokhale) [inline plugins/processors/round/README.md]: Does this need to be `transformation`? This seems to be the value used for other math processors

**Comment by alexgokhale** [comment]
@srebhan Think this should be ready for a re-review now 🙏

File: plugins/processors/round/README.md (hunk 10 above)
Comment (srebhan) [reply plugins/processors/round/README.md]: Maybe both.

File: plugins/processors/round/README.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  ## Round only numeric fields matching the filter criteria below.
+  ## Excludes takes precedence over includes.
+  # fields_include = []
+  # fields_exclude = []
+```
+
+## Example
+
+Round each value the _inputs.cpu_ plugin generates, except for the
+_usage\_steal_, _usage\_user_, _uptime\_format_, _usage\_idle_ field:
```
Comment (srebhan) [inline plugins/processors/round/README.md:42]:
```suggestion
`usage_steal`, `usage_user`, `uptime_format`, `usage_idle` field:
```

Comment (srebhan) [inline plugins/processors/round/README.md]:
```suggestion
`usage_steal`, `usage_user`, `uptime_format`, `usage_idle` field:
```

File: plugins/processors/round/round.go [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+func round[V constraints.Integer | constraints.Float](value V, factor float64) V {
+	return V(math.Round(float64(value)/factor) * factor)
+}
+
+func init() {
+	processors.Add("round", func() telegraf.Processor {
+		return &Round{
+			Precision: 3,
+		}
+	})
```
Comment (srebhan) [inline plugins/processors/round/round.go]: This doesn't match the documented default of `1` in `sample.conf`! How about making a precision of zero the default?

File: plugins/processors/round/round.go [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+		return round(v, p.factor)
+	default:
+		p.Log.Debugf("Value (%v) type invalid: [%v] is not an int, uint or float", v, reflect.TypeOf(value))
+	}
+	return value
+}
+
+func round[V constraints.Integer | constraints.Float](value V, factor float64) V {
+	return V(math.Round(float64(value)/factor) * factor)
+}
```
Comment (srebhan) [inline plugins/processors/round/round.go]: Please split the rounding for integers and floats! I've run into enough numerical issues to see that this is not a good idea. :-)

File: plugins/processors/round/round_test.go [hunk 14]
```diff
@@ -0,0 +1,363 @@
...
+					map[string]interface{}{"value": int64(-34437)},
+					time.Unix(0, 0),
+				),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			plugin := Round{
+				Precision: 2,
```
Comment (srebhan) [inline plugins/processors/round/round_test.go]: Please add the precision as a test parameter! This makes the test more versatile...

File: plugins/processors/round/round_test.go [hunk 15]
```diff
@@ -0,0 +1,363 @@
...
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+// Verifies that values are rounded correctly
+func TestRound_PositivePrecision(t *testing.T) {
```
Comment (srebhan) [inline plugins/processors/round/round_test.go]: Please follow the go naming conventions...
```suggestion
func TestPositivePrecision(t *testing.T) {
```

Same for the other functions below. Btw, if you add the `precision` as a test parameter, you can combine the positive and negative tests.

**Review by srebhan — COMMENTED** [review]
Thanks @alexgokhale! Almost there. Some more comments from my side.

**Comment by alexgokhale** [comment]
@srebhan Have actioned your comments - there's one test case which I'm not sure on the behaviour of:

Rounding `-34457` as an int with precision `-2` (i.e. factor 100) results in an output of `-34400`, because the value is being truncated when dividing as an integer. Is this the behaviour you would expect?

I would have expected to want to round half away from zero here, resulting in `-34500`, although this would require handling  part of the calculation as a float.

File: plugins/processors/round/round.go [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	v := value / f // -12.55
+
+	remainder := (value / (f / 10)) % 10
+	if remainder < 5 {
+		return v * f
+	}
+
+	return (v + 1) * f
```
Comment (srebhan) [inline plugins/processors/round/round.go]: I think your rounding is off for negative numbers... How about
```suggestion
	// Rounding to the full integer or a fraction will result
	// in the integer itself, so skip the computation.
	if factor < 10 {
		return value
	}

	// Compute relevant operators. As we need to round we
	// use an effective factor of one order of magnitude
	// less to keep the fractional part in the resulting
	// integer.
	f := factor / 10
	v := int64(value) / f
	r := v % 10

	// Round away from zero for positive and negative
	// values with an absolute fraction greater or
	// equal 1/2.
	if r <= -5 {
		return V((v - r - 10) * f)
	}
	if r >= 5 {
		return V((v - r + 10) * f)
	}

	// Floor the value as the absolute fraction is less
	// than 1/2.
	return V((v - r) * f)
```

File: plugins/processors/round/round.go [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+
+func roundFloat[V constraints.Integer | constraints.Float](value V, factor float64) V {
+	return V(math.Round(float64(value)/factor) * factor)
+}
+
+func init() {
+	processors.Add("round", func() telegraf.Processor {
+		return &Round{
+			Precision: 0,
+		}
```
Comment (srebhan) [inline plugins/processors/round/round.go]: Zero is the default value in go so we can use
```suggestion
		return &Round{}
```
here.

File: plugins/processors/round/round.go [hunk 18]
```diff
@@ -0,0 +1,119 @@
...
+	case uint32:
+		return roundInt(v, int64(p.factor))
+	case uint64:
+		return roundInt(v, int64(p.factor))
+	case float32:
+		return roundFloat(v, p.factor)
+	case float64:
+		return roundFloat(v, p.factor)
+	default:
+		p.Log.Debugf("Value (%v) type invalid: [%v] is not an int, uint or float", v, reflect.TypeOf(value))
```
Comment (srebhan) [inline plugins/processors/round/round.go]: Please don't spam the debug log. I agree we want this but maybe at trace level. Furthermore, there is no need for reflection here
```suggestion
	default:
		p.Log.Tracef("Invalid type %T for value '%v', value, value)
```

File: plugins/processors/round/round_test.go [hunk 19]
```diff
@@ -0,0 +1,359 @@
...
+			require.NoError(t, plugin.Init())
+
+			actual := plugin.Apply(tt.input...)
+			testutil.RequireMetricsEqual(t, tt.expected, actual)
+		})
+	}
+}
+
+// Verifies that round() returns zero values as 0
+func TestRoundWithZeroValue(t *testing.T) {
```
Comment (srebhan) [inline plugins/processors/round/round_test.go]: This whole test can be subsumed in the test above...

File: plugins/processors/round/round_test.go
Comment (srebhan) [inline plugins/processors/round/round_test.go:1]: Please also add a test for non-numeric fields like `bool` or `string` so we do have a defined behavior there, i.e. we keep those field values...

File: plugins/processors/round/sample.conf [hunk 20]
```diff
@@ -0,0 +1,11 @@
+# Round numerical fields
+[[processors.round]]
+  ## Precision to round to.
+  ## A positive number indicates rounding to the right of the decimal separator (i.e. the fractional part).
+  ## A negative number indicates rounding to the left of the decimal separator.
+  # precision = 0
+
+  ## Round only numeric fields matching the filter criteria below.
+  ## Excludes takes precedence over includes.
+  # fields_include = []
```
Comment (srebhan) [inline plugins/processors/round/sample.conf]: Maybe this makes the default behavior more clear?

```suggestion
  # fields_include = ["*"]
```

**Review by srebhan — COMMENTED** [review]
Nice! Thanks @alexgokhale! One last suggestion regarding the `field_include` parameter in the sample-configuration... Don't forget to run `make docs` afterwards...

File: plugins/processors/round/README.md [hunk 21]
```diff
@@ -0,0 +1,62 @@
+# Round Processor Plugin
+
+This plugin allows to round numerical field values to the configured precision.
+This is particularly useful in combination with the [dedup processor][dedup]
+to reduce the number of metrics sent to the output if only a lower precision
+is required for the values.
+
+⭐ Telegraf v1.36.0
+🏷️ math, transformation
```
Comment (srebhan) [inline plugins/processors/round/README.md]:
```suggestion
🏷️ transformation
```

**Review by srebhan — APPROVED** [review]
Thanks @alexgokhale! Fixing the tag to push the PR forward...

File: plugins/processors/round/round.go [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Round struct {
+	Precision     int             `toml:"precision"`
+	IncludeFields []string        `toml:"include_fields"`
+	ExcludeFields []string        `toml:"exclude_fields"`
```
Comment (skartikey) [inline plugins/processors/round/round.go:22]: Minor: The README shows `fields_include` and `fields_exclude`, but the struct uses `IncludeFields` and `ExcludeFields`. The TOML tags should match the documentation.

File: plugins/processors/round/round.go [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+	if r >= 5 {
+		return V((v - r + 10) * f)
+	}
+
+	// Floor the value as the absolute fraction is less
+	// than 1/2.
+	return V((v - r) * f)
+}
+
+func roundFloat[V constraints.Integer | constraints.Float](value V, factor float64) V {
```
Comment (skartikey) [inline plugins/processors/round/round.go]: This allows integers, but looking at the usage, only float32 and float64 call this function. The constraint should be:
```suggestion
func roundFloat[V constraints.Float](value V, factor float64) V {
```

**Review by skartikey — COMMENTED** [review]
@alexgokhale Great work on this PR, the type conversion is handled correctly, and the logic is functionally sound. Thanks for putting this together!

One small suggestion regarding code clarity: the current integer rounding logic works correctly, but it's a bit more complex than it needs to be. We might consider simplifying it by using `math.Round`, which also helps ensure consistent behavior across numeric types.

Here's a proposed refactor using generics to clean up the rounding logic:

```go
// Simplified rounding for integers only
func roundInt[V constraints.Integer](value V, factor float64) V {
	// For positive precision (factor < 1.0), integers don't need rounding
	if factor < 1.0 {
		return value
	}
	
	// Use standard math.Round for consistent behavior
	rounded := math.Round(float64(value) / factor) * factor
	return V(rounded)
}

// Simplified rounding for floats only  
func roundFloat[V constraints.Float](value V, factor float64) V {
	// Always round floats regardless of factor
	rounded := math.Round(float64(value) / factor) * factor
	return V(rounded)
}

// The round() method with corrected function calls:
func (p *Round) round(value interface{}) interface{} {
	switch v := value.(type) {
	case int:
		return roundInt(v, p.factor)
	case int8:
		return roundInt(v, p.factor)
	case int16:
		return roundInt(v, p.factor)
	case int32:
		return roundInt(v, p.factor)
	case int64:
		return roundInt(v, p.factor)
	case uint:
		return roundInt(v, p.factor)
	case uint8:
		return roundInt(v, p.factor)
	case uint16:
		return roundInt(v, p.factor)
	case uint32:
		return roundInt(v, p.factor)
	case uint64:
		return roundInt(v, p.factor)
	case float32:
		return roundFloat(v, p.factor)
	case float64:
		return roundFloat(v, p.factor)
	default:
		p.Log.Tracef("Invalid type %T for value '%v'", value, value)
		return value
	}
}
```

I haven’t tested this refactor, but in theory it should work as intended. Let me know your thoughts, happy to discuss or tweak the approach.

**Comment by alexgokhale** [comment]
@skartikey On the rounding logic &nbsp; @srebhan suggested keeping the logic for the two of these separate due to previous issues experienced with sharing the same rounding logic for floats and integers. Your suggestion makes the functions almost identical, is there a shared preference for whether to keep the logic for these distinct vs just using `math.Round`? I will then update to reflect that decision.

For `fields_include/exclude` it seems like some processors use `include_fields` (e.g. `noise`), but the recommendation here was to use `fields_include`. Which of these is the most up to date preference across telegraf? I'll then make the TOML keys, sample config, and variable names consistent with this.

**Comment by skartikey** [comment]
@alexgokhale Thanks for the context. I think let’s keep the rounding logic as it is for now, we can always revisit and improve it later if needed.

**Review by skartikey — APPROVED** [review]
@alexgokhale Nice work! Thanks for your contribution!

**Comment by alexgokhale** [comment]
Any update on this @mstrandboge? 🙏
